pkg/ccl/sqlccl/restore.go

```diff
@@ -722,20 +733,30 @@ func Restore(
 		for i := range newSpans {
 			span := newSpans[i]
 			g.Go(func() error {
-				req := &roachpb.AdminScatterRequest{
-					Span: roachpb.Span{Key: span.Key, EndKey: span.EndKey},
+				reqs := []roachpb.AdminScatterRequest{
+					{Span: roachpb.Span{Key: span.Key, EndKey: span.EndKey}},
 				}
-				res, pErr := client.SendWrapped(gCtx, db.GetSender(), req)
-				if pErr != nil {
-					return pErr.GoError()
-				}
-				// Scatter is best-effort, so log why any individual ranges
-				// didn't get scattered.
-				for _, r := range res.(*roachpb.AdminScatterResponse).Ranges {
-					if r.Error != nil {
-						log.Warningf(ctx, "error scattering range [%s,%s): %+v",
-							r.Span.Key, r.Span.EndKey, r.Error.GoError())
+				for {
+					if len(reqs) == 0 {
+						break
+					}
+					res, pErr := client.SendWrapped(gCtx, db.GetSender(), &reqs[len(reqs)-1])
+					if pErr != nil {
+						return pErr.GoError()
+					}
+					reqs = reqs[:len(reqs)-1]
+					// Scatter is best-effort, so log why any individual ranges
+					// didn't get scattered.
+					for _, r := range res.(*roachpb.AdminScatterResponse).Ranges {
+						if r.Error != nil {
+							log.Warningf(ctx, "error scattering range [%s,%s): %+v",
+								r.Span.Key, r.Span.EndKey, r.Error.GoError())
+							reqs = append(reqs, roachpb.AdminScatterRequest{
+								Span: r.Span,
+							})
+						}
 					}
+					time.Sleep(50 * time.Millisecond)
 				}
```

Comment:
A crappy implementation of repeatedly scattering all ranges until they report there's no work to be done.